src/actions/write/submitAuctionBid.ts

```diff
@@ -160,31 +163,45 @@ export const submitAuctionBid = (
       },
     };
 
-    // delete the auction
-    delete auctions[name];
-
     /**
      * Give the unsettled value to the protocol (it should already have the floor price from the initiated auction)
      * Deduct the unsettled final bid value from the caller
      * Return floor price from the protocol balance to the initiator, if necessary
      */
-    balances[SmartWeave.contract.id] =
-      (balances[SmartWeave.contract.id] || 0) + finalBidForCaller.valueOf();
-    balances[caller] = (balances[caller] || 0) - finalBidForCaller.valueOf();
+    updatedBalances[SmartWeave.contract.id] += finalBidForCaller.valueOf();
+    updatedBalances[caller] -= finalBidForCaller.valueOf();
 
     if (caller !== existingAuction.initiator) {
-      balances[existingAuction.initiator] =
-        (balances[existingAuction.initiator] || 0) + existingAuction.floorPrice;
-      balances[SmartWeave.contract.id] =
-        (balances[SmartWeave.contract.id] || 0) - existingAuction.floorPrice;
+      // pull in the intiators existing balance and update it
+      updatedBalances[existingAuction.initiator] =
+        (state.balances[existingAuction.initiator] || 0) +
+        existingAuction.floorPrice;
+      // use the most recent protocol balance and update it
+      updatedBalances[SmartWeave.contract.id] -= existingAuction.floorPrice;
     }
 
     // update the state
-    state.auctions = auctions; // NOTE: we don't need to reassign here as auctions is a reference
-    state.balances = balances; // NOTE: we don't need to reassign here as auctions is a reference
-    state.records = records; // NOTE: we don't need to reassign here as auctions is a reference
-    state.reserved = reserved; // NOTE: we don't need to reassign here as auctions is a reference
-    state.demandFactoring = tallyNamePurchase(demandFactoring);
+    const balances = {
+      ...state.balances,
+      ...updatedBalances,
+    };
+
+    // update our records
+    const records = {
+      ...state.records,
+      ...updatedRecords,
+    };
+
+    // remove the current name from auction
+    const { [name]: _, ...auctions } = state.auctions;
+
+    // update our state
+    Object.assign(state, {
+      auctions,
+      balances,
+      records,
+      demandFactoring: tallyNamePurchase(state.demandFactoring),
+    });
```

Comment:
we could move this up to `contract.ts` being responsible for the final assign and just return specific values here - similar to how we are doing it in tickInternal.
 
the return type for `submitAuctionBid` could then be `Pick<IOState, auctions, reserved, balances, records>`